sql/payday.sql

```diff
@@ -204,30 +205,37 @@ CREATE TRIGGER process_payment_instruction BEFORE UPDATE OF is_funded ON payday_
     WHEN (NEW.is_funded IS true AND OLD.is_funded IS NOT true)
     EXECUTE PROCEDURE process_payment_instruction();
 
--- Create a trigger to process takes
 
-CREATE OR REPLACE FUNCTION process_take() RETURNS trigger AS $$
+-- Create a trigger to process distributions based on takes
+
+CREATE OR REPLACE FUNCTION process_distribution() RETURNS trigger AS $$
     DECLARE
-        actual_amount numeric(35,2);
-        team_balance numeric(35,2);
+        amount      numeric(35,2);
+        balance_    numeric(35,2);
+        available_  numeric(35,2);
     BEGIN
-        team_balance := (
-            SELECT new_balance
-              FROM payday_participants
-             WHERE username = NEW.team
-        );
-        IF (team_balance <= 0) THEN RETURN NULL; END IF;
-        actual_amount := NEW.amount;
-        IF (team_balance < NEW.amount) THEN
-            actual_amount := team_balance;
+        amount := NEW.amount;
+
+        balance_ := (SELECT balance FROM payday_teams WHERE id = NEW.team_id);
+        IF balance_ < amount THEN
+            amount := balance_;
+        END IF;
+
+        available_ := (SELECT available FROM payday_teams WHERE id = NEW.team_id);
+        IF available_ < amount THEN
+            amount := available_;
+        END IF;
+
+        IF amount > 0 THEN
+            UPDATE payday_teams SET available = (available - amount) WHERE id = NEW.team_id;
```

Comment:
So many varying values here :/ Both balance *and* available are reducing at the same rate - we decrease the `available` here, and `pay` ends up decreasing the `balance` by the same amount. Wondering if there is a cleaner way to model this... Not thought this through, just speaking out loud - will revisit once I'm done with the rest